subprojects/gradle-plugin/src/main/java/org/spongepowered/gradle/vanilla/internal/repository/IvyModuleWriter.java

```diff
@@ -155,11 +155,26 @@ private void writeDependencies(final Set<GroupArtifactVersion> dependencies) thr
     }
 
     private void writeDependency(final GroupArtifactVersion dep) throws XMLStreamException {
-        this.writer.writeEmptyElement("dependency");
+        boolean hasClassifier = dep.classifier() != null;
+
+        if (hasClassifier) {
+            this.writer.writeStartElement("dependency");
+        } else {
+            this.writer.writeEmptyElement("dependency");
+        }
+
         this.writer.writeAttribute("org", dep.group());
         this.writer.writeAttribute("name", dep.artifact());
         this.writer.writeAttribute("rev", dep.version());
         this.writer.writeAttribute("transitive", "false");
+
+        if (hasClassifier) {
+            this.writer.writeEmptyElement("artifact");
+            this.writer.writeAttribute("name", dep.artifact());
+            this.writer.writeAttribute(IvyModuleWriter.VANILLAGRADLE + ":classifier", dep.classifier());
```

Comment:
Gradle reads the `classifier` option here, but it's a *gradle-defined* option, not a VG-defined option, so it shouldn't have the VG namespace